File: backend/interview_portal/src/main/java/by/interview/portal/dto/CalendarDTO.java
package by.interview.portal.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Set;

import by.interview.portal.domain.ExcludedTimeSlot;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CalendarDTO {

    private List<SpecifiedTimeDTO> specifiedTimeDTOs;

    private List<ExcludedTimeSlot> excludedTimeSlots;

    private Set<InterviewDTO> interviews;
}

File: frontend/src/app/api/models/calendar-dto.ts
/* tslint:disable */
import { ExcludedTimeSlot } from './excluded-time-slot';
import { SpecifiedTimeDTO } from './specified-time-dto';
import { InterviewDTO } from './interview-dto';

export interface CalendarDTO {

  excludedTimeSlots?: ExcludedTimeSlot[];

  interviews?: InterviewDTO[];

  specifiedTimeDTOs?: SpecifiedTimeDTO[];
}
